store/storetest/post_store.go

```diff
@@ -1352,27 +1352,92 @@ func testPostCountsByDay(t *testing.T, ss store.Store) {
 	_, err = ss.Post().Save(o2a)
 	require.Nil(t, err)
 
+	bot1 := &model.Bot{
+		Username:    "username",
+		Description: "a bot",
+		OwnerId:     model.NewId(),
+		UserId:      model.NewId(),
+	}
+	_, err = ss.Bot().Save(bot1)
+
+	b1 := &model.Post{}
+	b1.Message = "bot message one"
+	b1.ChannelId = c1.Id
+	b1.UserId = bot1.UserId
+	b1.CreateAt = utils.MillisFromTime(utils.Yesterday())
+	_, err = ss.Post().Save(b1)
+	require.Nil(t, err)
+
+	b1a := &model.Post{}
+	b1a.Message = "bot message two"
+	b1a.ChannelId = c1.Id
+	b1a.UserId = bot1.UserId
+	b1a.CreateAt = utils.MillisFromTime(utils.Yesterday()) - (1000 * 60 * 60 * 24 * 2)
+	_, err = ss.Post().Save(b1a)
+	require.Nil(t, err)
+
 	time.Sleep(1 * time.Second)
 
-	if r1, err := ss.Post().AnalyticsPostCountsByDay(t1.Id); err != nil {
+	// summary of posts
+	// yesterday - 2 non-bot user posts, 1 bot user post
+	// 3 days ago - 2 non-bot user posts, 1 bot user post
+
+	// last 31 days, all users (including bots)
+	if r1, err := ss.Post().AnalyticsPostCountsByDay(t1.Id, false, false); err != nil {
```

Comment:
nit: could we start using assert/require to simplify the testing code?

Reply:
yes, I think this is another good suggestion and not really a nit.  Also removing the `rowX` variables and accessing the array elements in the asserts reduces this section of code down to 36 lines, from 56.

```go
row1 := r1[0]
if row1.Value != 3 {
	t.Fatal(row1)
}
```
```go 
assert.Equal(t, float64(1), r1[0].Value)
```